Plugins/NativeEngine/Include/Babylon/ShaderCache.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <string_view>
+#include <memory>
+
+namespace Babylon
+{
+    class ShaderCache
+    {
+    public:
+        ShaderCache();
+        ~ShaderCache();
+
+        void SerializeOutTo(std::ofstream& stream);
+        uint32_t SerializeInFrom(std::ifstream& stream);
+        
+        class Impl;
+        static Impl* GetImpl();
```

Comment:
I'm a bit confused on this. I thought Impl was supposed to be a way of hiding the implementation details from the header and eliminating webs of header dependencies, but in this case it seems like Impl is part of the public contract?

Reply:
I've reworked it a bit.